### PR TITLE
feat(math): add mutative api for BigDec.BigInt()

### PR DESCRIPTION
**Pull request by DongLieu**
Similar to [PR17803](https://github.com/cosmos/cosmos-sdk/pull/17803) I think we should also have such an API

### CODE REVIEW

File: osmomath/decimal.go [hunk 1]
```diff
@@ -248,6 +248,15 @@ func (d BigDec) BigInt() *big.Int {
 	return cp.Set(d.i)
 }
 
+// BigIntMut converts BigDec to big.Int, mutative the input
+func (d BigDec) BigIntMut() *big.Int {
```
Comment (mattverse) [inline osmomath/decimal.go:252]: I think ToBigInt is not a suitable method name here, recommending change to the following:
```suggestion
func (d BigDec) ToBigInt() *big.Int {
```

**Review by mattverse — CHANGES_REQUESTED** [review]
mostly looks good, requested change for method name change :)

**Comment by mattverse** [comment]
Please ping me once r4r

**Comment by DongLieu** [comment]
Ok

**Comment by DongLieu** [comment]
Thanks you

**Comment by DongLieu** [comment]
I'm waiting for a PR https://github.com/cosmos/cosmos-sdk/pull/18576 at cosmos hub. Give me more time

**Pull request closed without merging**